### PR TITLE
Remove jQuery `ready` usage

### PR DESCRIPTION
**Pull request by silverwind**
Replace it with equal function of our own and enable the eslint rule to forbid future usage.

### CODE REVIEW

File: web_src/js/utils/dom.js [hunk 1]
```diff
@@ -67,3 +67,11 @@ export function hideElem(el) {
 export function toggleElem(el, force) {
   elementsCall(el, toggleShown, force);
 }
+
+export async function onDomReady(cb) {
```
Comment (silverwind) [inline web_src/js/utils/dom.js]:
```suggestion
export function onDomReady(cb) {
```

**Comment by wxiaoguang** [comment]
I guess we do not need `DOMContentLoaded`.

Because the HTML code looks like this:

```
<html>
<body>
  <div page>
      ....
  </div>

  <script index.js>
</body>
</html>
```

When `index.js` gets executed, all necessary DOM elements are already there (ready), we do not need to wait for the remaining `</body>`.

So, how about making code just call every init function directly in `index.js`?

**Comment by silverwind** [comment]
Are you 100% certain this will work? I'm trying to play it safe here.

**Comment by wxiaoguang** [comment]
> Are you 100% certain this will work? I'm trying to play it safe here.

99.9999% sure.

`DOMContentLoaded / $.ready` is only used for 2 purposes:

* Some historical broken browser (as old as IE6-9), if the DOM is not ready, many operations on DOM cause page crash.
     It's never been a problem for modern browsers.
* For JS code which is ahead of its related DOM elements, like `<script>$.ready</script><div the-related></div>`
     It's not our case.

In fact, we already have a lot of code operating the DOM directly without DOM ready, for example, the "Clone" / "DiffView" inline `<script>`.


I think it's safe to simplify the code in 1.20, we have enough time to eat our own dogfood.

**Comment by silverwind** [comment]
Ok, then let's try it. If there are regressions, we should be able to notice in time.

**Comment by silverwind** [comment]
Tried it, got this error:

image

So I think we can not do it, at least not without further refactors.

**Comment by silverwind** [comment]
Also, vue components are broken without the callback:

image

**Comment by wxiaoguang** [comment]
Hmm .... it's caused by `<script module>` .....

Tricky .... I have a feeling that it's not `DOMContentLoaded`'s problem, but the `<script module>` is wrong, it wrongly defers its execution.

So to keep things simple, let's keep the `DOMContentLoaded`, I will take a look at the `<script module>` problem later.

**Comment by silverwind** [comment]
Yeah, so PR is ready as-is.

**Review by delvh — APPROVED** [review]

**Review by lunny — APPROVED** [review]

**Comment by techknowlogick** [comment]
🤖 🎺

**Pull request merged**